tracer/src/Datadog.Trace/DiagnosticListeners/EndpointFeatureProxy.cs

```diff
@@ -15,12 +15,13 @@ namespace Datadog.Trace.DiagnosticListeners
     /// </summary>
     [Browsable(false)]
     [EditorBrowsable(EditorBrowsableState.Never)]
-    public interface IEndpointFeature
+    public class EndpointFeatureProxy
```

Comment:
I'm guessing that we can't make this proxy a struct because we have to call a method on the real object, is that correct?

Reply:
Yep, exactly - we can't use `[DuckCopy]` structs if we have to call a method, so it has to be interface, abstract class, or virtual class. We've typically favoured interfaces for simplicity, but @tonyredondo did some benchmarks that show virtual classes are fastest, so we should favour them where possible (and I don't think there's really any case where we _need_ to use an interface over a virtual class)